datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -3813,3 +3813,58 @@ logical_plan
 01)SubqueryAlias: b
 02)--Projection: Int64(1) AS a
 03)----EmptyRelation
+
+
+statement ok
+set datafusion.execution.target_partitions = 1;
+
+statement ok
+set datafusion.explain.logical_plan_only = false;
+
+# Right Hash Joins preserve the right ordering
+query IIII
+SELECT * FROM (
+    SELECT 1 as a, 1 as b
+) as lhs RIGHT JOIN (
```

Comment:
It may be worth to increase test coverage by adding more input data to this test -- this version doesn't seem to cover last loop of `append_probe_indices_in_order` (the `for value in prev_index..range.end` one), and probably it's also worth checking cases with missing probe index in the middle of the batch / no nulls on build side for the batch.

Reply:
@berkaysynnada, let's update the test to address